### PR TITLE
feat: support TypeScript constructor params

### LINKED ISSUE #192: Parameter properties are ignored
Typescript has [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties), but `deno_doc` ignores them. To reproduce, compare the output between running `deno doc` on this:

```typescript
class Example {
  name: string

  constructor(name: string) {
    this.name = name
  }
}
```

vs this

```typescript
class Example {
  constructor(
    public name: string
  ) {}
}
```

`deno doc` will output the `name` property for the first example but not for the second.

### PR DESCRIPTION
**Pull request by kitsonk**
Closes #192

This is an alternative to #196. Like the recent has_body change, instead of inferring, the doc nodes returned more reflect the code as written, and it is up to the printers to determine how to display that to the user.

In this situation, printers need to decide how to display a `public` constructor param, along with the other accessibility keywords, override and readonly, which is to likely reflect them in the constructor signature as well as the properties documented on the class.

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM

File: lib/types.d.ts [hunk 1]
```diff
@@ -95,13 +95,19 @@ export interface ClassDef {
   decorators?: DecoratorDef[];
 }
 
+export type ClassConstructorParamDef = ParamDef & {
```
Comment (dsherret) [inline lib/types.d.ts:98]: Nitpick: seems more consistent to use an interface?

**Review by dsherret — APPROVED** [review]
LGTM too

File: lib/types.d.ts (hunk 1 above)
Comment (kitsonk) [reply lib/types.d.ts:98]: You can't here, because `ParamDef` is a tagged union type alias and interfaces can't extend type aliases.

**Pull request merged**